app/lib/account/agent.dart

```diff
@@ -12,41 +12,74 @@ import 'models.dart';
 final _uuidRegExp =
     RegExp(r'^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$');
 
+/// Agents identify authenticated users or automated system accounts.
+///
+/// The format and/or the content prefix identifies the type of the agent.
+/// Newer agent identifiers must contain a unique (but maybe opaque)
+/// part that corresponds with an internal or external authentication
+/// identifier (like email address or repository URL).
+///
+/// Older identifiers may only describe the type of the agent.
 abstract class KnownAgents {
-  static const githubActions = 'service:github-actions';
-  static const gcpServiceAccount = 'service:gcp-service-account';
+  /// Non-specific agent - only specifies it is from GitHub Actions.
+  static const genericGithubActions = 'service:github-actions';
+
+  /// Non-specific agent - only specifies it is from GCP Service Account.
+  static const genericGcpServiceAccount = 'service:gcp-service-account';
+
   // Agent for pub admin actions.
   static const pubSupport = 'support@pub.dev';
 
-  static const _values = <String>{
-    githubActions,
-    gcpServiceAccount,
+  /// Returns an agentId in the format of `service:github-actions:<repositoryOwnerId>/<repositoryId>`
+  static String githubActionsAgentId({
+    required String repositoryOwnerId,
+    required String repositoryId,
+  }) {
+    return [genericGithubActions, '$repositoryOwnerId/$repositoryId'].join(':');
```

Comment:
nit: feel free to ignore :D

```suggestion
    return '$_GithubActionsPrefix:$repositoryOwnerId/$repositoryId';
```

and then have:
```
/// Prefix for `agentId`s on the form:
/// `service:github-actions:<repositoryOwnerId>/<repositoryId>`
final _GithubActionsPrefix = 'service:github-actions'
```

because `genericGithubActions` indicates a generic agentId, not a prefix.
And for all intents and purposes `genericGithubActions`is a historical value, not a prefix.

--------

Or even simpler, just hardcode it:
```suggestion
    return 'service:github-actions:$repositoryOwnerId/$repositoryId';
```

Reply:
Updated with the separate field names (I think it is better as we also need to repeat the prefix in another place).